app/cdap/components/Administration/TetheringTabContent/NewTetheringRequest/index.tsx

```diff
@@ -0,0 +1,206 @@
+  }
+`;
+
+const Divider = styled.span`
+  font-size: 1.25rem;
+  padding: 0 5px;
+`;
+
+const StyledBodyContainer = styled(BodyContainer)`
+  padding: 30px;
+  flex-direction: column;
+  align-items: flex-start;
+`;
+
+const ButtonsContainer = styled.div`
+  margin-left: -30px;
+`;
+
+interface IValidationErrors {
+  instanceName?: IErrorObj;
+  projectName?: IErrorObj;
+  region?: IErrorObj;
+}
+
+const NewTetheringRequest = () => {
+  const [namespace, setNamespace] = useState('');
+  const [cpuLimit, setCpuLimit] = useState('');
+  const [memoryLimit, setMemoryLimit] = useState('');
+  const [projectName, setProjectName] = useState('');
+  const [region, setRegion] = useState('');
+  const [instanceName, setInstanceName] = useState('');
+  const [showAlert, setShowAlert] = useState(false);
+  const [apiErr, setApiErr] = useState({} as IApiError);
+  const [validationErrors, setValidationErrors] = useState({} as IValidationErrors);
+  const history = useHistory();
+
+  const stateUpdater = {
+    name: setNamespace,
+    cpuLimit: setCpuLimit,
+    memoryLimit: setMemoryLimit,
+    projectName: setProjectName,
+    region: setRegion,
+    instanceName: setInstanceName,
+  };
+
+  const resetValues = () => {
+    setNamespace('');
+    setCpuLimit('');
+    setMemoryLimit('');
+    setProjectName('');
+    setRegion('');
+    setInstanceName('');
+    setShowAlert(false);
+    setApiErr({} as IApiError);
+  };
+
+  const areInputsValid = (): boolean => {
+    const requiredFields = [
+      { name: 'projectName', val: projectName },
+      { name: 'region', val: region },
+      { name: 'instanceName', val: instanceName },
+    ];
+    let allValid = true;
+    const errors = {};
+
+    requiredFields.forEach((field) => {
+      let errObj = {};
+      if (!field.val) {
+        errObj = {
+          msg: T.translate(`${I18NPREFIX}.validationError`, { fieldName: field.name }),
+        };
+        allValid = false;
+      }
+      errors[field.name] = errObj;
+    });
+
+    setValidationErrors((prevValidationErrors) => ({
+      ...prevValidationErrors,
+      ...errors,
+    }));
+
+    return allValid;
+  };
+
+  const handleChange = (target: string, updatedVal: string | number) => {
+    stateUpdater[target](updatedVal);
+  };
+
+  const handleSend = async () => {
+    if (areInputsValid()) {
+      const connectionInfo = {
+        peer: instanceName,
+        endpoint: 'http://www.google.com', // TODO: needs clarification on how to obtain
+        namespaceAllocations: [{ namespace, cpuLimit, memoryLimit }],
+        metadata: {
+          project: projectName,
+          location: region,
+        },
+      };
+
+      try {
+        await TetheringApi.createTethering({}, connectionInfo).toPromise();
+      } catch (err) {
+        setApiErr(err);
+      }
+      setShowAlert(true);
+    }
+  };
+
+  const handleCancel = () => {
+    history.push('/administration/tethering');
+  };
+
+  const handleCloseAlert = () => {
```

Comment:
this method is not required, you can directly pass 'resetValues' as onClose handler